src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -184,10 +185,25 @@ const usePushNotificationHook = () => {
             id: notification?.data?.channel_id,
             channelType: 'PM'
           });
+
           return fetchChannelDetail(channel);
         }
 
-        const channel = await ChannelList.getSchemaById(localDb, notification?.data?.channel_id);
+        let channel = await ChannelList.getSchemaById(localDb, notification?.data?.channel_id);
+        if (!channel) {
+          try {
+            setIsLoadingFetchingChannelDetail(true);
+            const fetchChannel = new ChannelList({
+              id: notification?.data?.channel_id,
+              channelType: 'PM'
+            });
+            await fetchChannelDetail(fetchChannel);
+            channel = await ChannelList.getSchemaById(localDb, notification?.data?.channel_id);
+          } finally {
+            setIsLoadingFetchingChannelDetail(false);
+          }
+        }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for fetching channel details seems to be well implemented. However, there is a potential issue with error handling. If an error occurs during the `fetchChannelDetail(fetchChannel)` call, it will not be caught and could potentially crash the application. To improve this, consider adding a catch block to handle any errors that may occur during the fetch operation.
```diff
          try {
            setIsLoadingFetchingChannelDetail(true);
            const fetchChannel = new ChannelList({
              id: notification?.data?.channel_id,
              channelType: 'PM'
            });
            await fetchChannelDetail(fetchChannel);
            channel = await ChannelList.getSchemaById(localDb, notification?.data?.channel_id);
          } 
+            catch (error) {
+              console.error('Error fetching channel detail:', error);
+              // Handle the error appropriately here
+            }
          finally {
            setIsLoadingFetchingChannelDetail(false);
          }
        }
```
This way, if an error occurs during the fetch operation, it will be logged and can be handled appropriately, preventing the application from crashing.


<!-- This is an auto-generated comment by OSS CodeRabbit -->